packages/db-ivm/src/operators/groupBy.ts

```diff
@@ -210,48 +210,64 @@ export function avg<T>(
   }
 }
 
+type CanMinMax = number | Date | bigint
+
 /**
  * Creates a min aggregate function that computes the minimum value in a group
- * @param valueExtractor Function to extract a numeric value from each data entry
+ * @param valueExtractor Function to extract a comparable value from each data entry
  */
-export function min<T, V extends number | Date | bigint>(
-  valueExtractor: (value: T) => V = (v) => v as unknown as V
-): AggregateFunction<T, V, V> {
+export function min<T extends CanMinMax>(): AggregateFunction<
+  T,
+  T | undefined,
+  T | undefined
+>
+export function min<T, V extends CanMinMax>(
+  valueExtractor: (value: T) => V
+): AggregateFunction<T, V | undefined, V | undefined>
+export function min<T, V extends CanMinMax>(
+  valueExtractor?: (value: T) => V
+): AggregateFunction<T, V | undefined, V | undefined> {
+  const extractor = valueExtractor ?? ((v: T) => v as unknown as V)
   return {
-    preMap: (data: T) => valueExtractor(data),
+    preMap: (data: T) => extractor(data),
     reduce: (values) => {
       let minValue: V | undefined
       for (const [value, _multiplicity] of values) {
-        if (minValue && value < minValue) {
-          minValue = value
-        } else if (!minValue) {
+        if (!minValue || !value || value < minValue) {
           minValue = value
```

Comment:
Is it correct to add `!value` here? It wasn't in the original if test and this means that if we have an existing `minValue` but `value = undefined` (or `value = 0` because `!0 === true`) we will overwrite the `minValue` with `undefined`/`0`.

Reply:
Should it be `if (!minValue || (value && value < minValue))` ?

Reply:
These comments also apply to `max`